src/main/java/org/elasticsearch/plugin/WordDelimiterPlugin.java

```diff
@@ -1,37 +1,40 @@
 package org.elasticsearch.plugin;
 
-import org.elasticsearch.plugins.AbstractPlugin;
-import org.elasticsearch.index.analysis.AnalysisModule;
+import org.elasticsearch.index.analysis.TokenFilterFactory;
+import org.elasticsearch.indices.analysis.AnalysisModule;
+import org.elasticsearch.plugins.AnalysisPlugin;
 import org.elasticsearch.common.component.LifecycleComponent;
 import org.elasticsearch.module.WordDelimiterService;
-import org.skroutz.elasticsearch.index.analysis.WordDelimiterBinderProcessor;
+import org.elasticsearch.plugins.Plugin;
 
+import java.util.ArrayList;
 import java.util.Collection;
-import static org.elasticsearch.common.collect.Lists.newArrayList;
+import java.util.Collections;
+import java.util.Map;
 
-public class WordDelimiterPlugin extends AbstractPlugin {
-  @SuppressWarnings("rawtypes")
-  private final Collection<Class<? extends LifecycleComponent>> services = newArrayList();
+import org.skroutz.elasticsearch.index.analysis.WordDelimiterTokenFilterFactory;
+
+public class WordDelimiterPlugin extends Plugin implements AnalysisPlugin {
+
+  private final Collection<Class<? extends LifecycleComponent>> services =
+          new ArrayList();
 
   public WordDelimiterPlugin() {
     services.add(WordDelimiterService.class);
   }
 
-  public String name() {
-    return "dynamic-word-delimiter";
+  @Override
+  public Collection<Class<? extends LifecycleComponent>> getGuiceServiceClasses() {
```

Comment:
I didn't find any documentation on this. We're supposed to deduce that from the API I guess, as it was the only method around services in Plugin class.